### ISSUE TITLE
Creating custom RKE cluster failed => Error: Cluster ID c-497hl: Pre-pulling kubernetes images

### ISSUE THREAD

**Issue by Edge94** [rancher/terraform-provider-rancher2#845]
Hi,

When creating a custom RKE cluster using Terraform and rancher2 provider the terraform script fails because of network latency.
But, the cluster itself continues deploying.
We are using an Artifactory proxy to download docker images from Docker Hub and a lot of people in the company are using it :-)

It seems to be a bug in the "rancher2_cluster_sync" resource.
I think the "rancher2_cluster_sync" resource must wait the cluster is in an active state and has downloaded all needed images.
If I wait a few minutes before running it again, the terraform script finished successfully.

I'm using :
- Rancher v2.5.10
- Terraform v0.13.5
- rancher2 provider v1.22.2

Terraform logs :
```
...
4317module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [9m10s elapsed]
4318module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [9m20s elapsed]
4319module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [9m30s elapsed]
4320module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [9m40s elapsed]
4321module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [9m50s elapsed]
4322module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [10m0s elapsed]
4323module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [10m10s elapsed]
4324module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [10m20s elapsed]
4325module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [10m30s elapsed]
4326module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [10m40s elapsed]
4327module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [10m50s elapsed]
4328module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [11m0s elapsed]
4329module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [11m10s elapsed]
4330module.k8s_openstack_rke.rancher2_cluster_sync.wait_cluster: Still creating... [11m20s elapsed]
4331Error: Cluster ID c-497hl: Pre-pulling kubernetes images
4332  on modules/k8s-openstack-rke/main.tf line 1318, in resource "rancher2_cluster_sync" "wait_cluster":
43331318: resource "rancher2_cluster_sync" "wait_cluster" {
```

Terraform script :
```
...
resource "rancher2_cluster_sync" "wait_cluster" {
  cluster_id =  rancher2_cluster.cluster.id
  state_confirm = 2
  
  timeouts {
    create = "30m"
  }
}
...
```

Thank you.

**Comment by Edge94** [comment]
Hi,

Any suggestion ?

Thanks

**Comment by rawmind0** [comment]
Hi @Edge94 , what it seems is that the rke deployment is failing when tries to pre pull the kubernetes images. The `rancher2_cluster_sync` should wait until the cluster is active or it's returning an error. Is this config working fine when executed from UI??

**Comment by Edge94** [comment]
Hi @rawmind0, in fact, the rke deployment is not failing. Only the rancher2_cluster_sync terraform resource, while waiting for the cluster to be active. If I run again the terraform script, it succeeds.

From UI, the cluster is deploying fine.

**Comment by rawmind0** [comment]
Yeah, the deployment is not failing at the end due to rancher is retrying. My question about the UI is more related with the cluster status until it's active. Is the cluster at some point in `Error` state at UI before got active by the rancher retry??

**Comment by Edge94** [comment]
I think so !

**Comment by rawmind0** [comment]
If the cluster is getting `error` state and just got `active` by the rancher retry, not so much can be done from here. The `rancher2_cluster_sync` is waiting for cluster active but failing if error is returned.

**Comment by Edge94** [comment]
Could we add a retry in rancher2_cluster_sync after getting an error state from the cluster ?

**Comment by rawmind0** [comment]
I don't think this gonna solve your issue, how many retries?? Would be enough with just one?? 

I think would be easier that you try to improve network latency or you can pre pull docker images in your hosts before running rke command. You may use  [rancher-images.txt](https://github.com/rancher/rancher/releases/download/v2.5.10/rancher-images.txt) to pre pull required images on your hosts, if the network latency is high

**Comment by Edge94** [comment]
Hi @rawmind0, I'm not able to improve the network latency because it's managed by network admins. And I think the main network latency is about a docker hub proxy that is overloaded.

Pre-pulling images is painful because we need to update the images list each time we have a new version.

I suggest adding a new parameter to set the retries number. By default, no retry (current behavior).

**Comment by matttrach** [comment]
This issue is a couple of years old now and can only reference unsupported versions, so I am going to close it, feel free to open a new issue if this is still an problem.
